heron/downloaders/src/java/com/twitter/heron/downloader/DLDownloader.java

```diff
@@ -0,0 +1,81 @@
+//  Copyright 2017 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.downloader;
+
+import java.io.File;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.function.Supplier;
+
+import com.google.common.annotations.VisibleForTesting;
```

Comment:
@cckellogg I can remove direct dependency. however the dlog/bk library depends on guava. I can try to shade them in dlog library. does that work for you?